src/main/webapp/scripts/keywords.js

```diff
@@ -14,4 +14,32 @@ $().ready(function() {
     $('#blob-input').addClass('hide').val(null);
     $('#form-title *').text('Copy Your Tweet Here');
   });
+
+  $('#keyword-form').on('submit', function(event) {
+    // Check whether the inputs are valid before submitting.
+    if ($('#blob-input').hasClass('hide')) {
+      // Check the text input
+      if ($('#text-input').val() == null || $('#text-input').val() === "") {
+        alert('Your input cannot be empty!'); 
+        event.preventDefault();
+        return false;
+      }
+    } else {  // Check blob input
```

Comment:
Cleaner comments if you mention this block checks if any file uploaded, and in the next block you check if a valid image file is uploaded.